packages/babylon-ts-sdk/docs/pegin-flow.md

```diff
@@ -0,0 +1,172 @@
+# Peg-in Flow
+
+This guide walks through creating a peg-in transaction step by step.
+
+## Overview
+
+The peg-in flow deposits BTC into a vault. The process involves:
+
+1. **Build the unsigned transaction** using `buildPeginPsbt()`
+2. **Fund the transaction** by adding UTXOs and calculating fees
+3. **Sign** the transaction via your wallet
+4. **Broadcast** to the Bitcoin network
+
+## Step 1: Build Unsigned Peg-in Transaction
+
+Use `buildPeginPsbt()` to create the vault output. This returns an **unfunded** transaction with one output (the vault address).
+
+```typescript
+import { buildPeginPsbt } from '@babylonlabs-io/ts-sdk/tbv/core/primitives';
+
+const peginResult = await buildPeginPsbt({
+  depositorPubkey: 'a1b2c3d4e5f6...',
+  claimerPubkey: 'b2c3d4e5f6a1...',
+  challengerPubkeys: ['c3d4e5f6a1b2...'],
+  pegInAmount: 100000n,
+  network: 'testnet',
+});
+
+console.log('Vault address:', peginResult.vaultScriptPubKey);
```

Comment:
BTC Vaults don't have an address. I suppose what you mean here is the BTC Vault Bitcoin script?